unyt/_deprecation.py

```diff
@@ -0,0 +1,26 @@
+import warnings
+from typing import Optional
+
+
+class UnytDeprecationWarning(UserWarning):
+    pass
```

Comment:
Can we use the built-in `DeprecationWarning` instead? We already use that elsewhere.